http_server/request_testing/request_listener.go

```diff
@@ -0,0 +1,54 @@
+package requesttesting
+
+import (
+	"net"
+	"errors"
+	"log"
+)
+
+type mockListener struct {
+	connChannel chan net.Conn
+	serverEndpoint net.Conn
+}
+
+// For each request, we create an HTTP listener that passes the request provided
+// as a function parameter to the server.
+func newMockListener(endpoint  net.Conn) mockListener {
+
+	channel := make(chan net.Conn, 1)
+	channel <- endpoint
```

Comment:
naming a variable after its type is not very telling, if the variable is short-lived you can shorten the name to be a single letter.